### ISSUE TITLE
Add REST event sequence discussion

### ISSUE THREAD

**Issue by dcmaf** [addyosmani/backbone-fundamentals#314]
The latest version of Chapter 3 introduces Backbone's RESTful API (fetch, save, destroy, etc.) and Chapter 5 provides a RESTful application tutorial. It would be helpful to follow-up on these with a discussion of what happens and when during these operations (e.g., validation, url construction, events, sync call, success/error callbacks, parsing, model state changes). Recommended location in or adjacent to the RESTful tutorial.

**Comment by addyosmani** [comment]
> It would be helpful to follow-up on these with a discussion of what happens and when during these operations (e.g., validation, url construction, events, sync call, success/error callbacks, parsing, model state changes). 

I agree. How detailed would you suggest the walkthrough here to be? It could be done as a bullet-point walkthrough or something more detailed.

**Comment by dcmaf** [comment]
I was thinking of something along the lines of sequence diagrams. Ideally, for all the CRUD operations (with variants for the failure cases).
